dstep/translator/Context.d

```diff
@@ -204,3 +225,66 @@ bool isGlobal(Cursor cursor)
 {
     return cursor.semanticParent.kind == CXCursorKind.CXCursor_TranslationUnit;
 }
+
+/**
+  * The collectTypeNames function scans the whole AST of the translation unit and produces
+  * a set of the type names in global scope.
+  *
+  * The type names are required for parsing C code (e.g. macro definition bodies),
+  * as C grammar isn't context free.
+  */
```

Comment:
To parse C-style casts (and `sizeof`s) you need to have information about type names. Without it you do not know if `(int)(4)` is a cast or a function call. This function collects that info.